pipeline/pipeline.py

```diff
@@ -0,0 +1,58 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+from task import AbstractTask
+
+
+@dataclass
+class PipelineConfig:
+    """Configuration of the workspace migration pipeline."""
+    base_dir: str
+    session: str
+
+
+class Pipeline:
+    """Class that coordinates run of a group of tasks, which form a DAG based on dependencies
+    defined in add_task().
+
+    The pipeline has built-in checkpoint i.e it will skip all complete tasks upon restart."""
+
+    @dataclass
+    class Node:
+        task: AbstractTask = None
+        children = []
```

Comment:
can you leave a comment for children? Is this the list of tasks that are executed after this Node(task) is done?

Reply:
Done. PTAL